assisted-inject-processor/src/main/java/com/squareup/inject/assisted/processor/AssistedInjectProcessor.kt

```diff
@@ -118,12 +125,6 @@ class AssistedInjectProcessor : AbstractProcessor() {
     val constructors = type.enclosedElements
         .filter { it.kind == CONSTRUCTOR }
         .cast<ExecutableElement>()
-    if (constructors.size == 1) {
-      // Only one constructor (common case). Use it! This also allows better error messages to be
-      // produced in the case where you have no @Assisted parameters on this constructor or are
-      // missing @AssistedInject.
-      return constructors.single()
```

Comment:
Now that we require `@AssistedInject` on the constructor this optimization had no real effect.